pkg/kp/rcstore/consul_store.go

```diff
@@ -301,7 +312,17 @@ func (s *consulStore) Watch(rc *fields.RC, quit <-chan struct{}) (<-chan struct{
 	errors := make(chan error)
 	input := make(chan *api.KVPair)
 
-	go consulutil.WatchSingle(kp.RCPath(rc.ID.String()), s.kv, input, quit, errors)
+	go func() {
+		rcp, err := RCPath(rc.ID)
+		if err != nil {
+			select {
+			case <-quit:
+			case errors <- err:
+			}
+			return
```

Comment:
I suppose I'm not concerned with releasing the RC, because if this agent has trouble handling the RC, I have no faith that another host will do any better.  Bubbling the error up to the main farm loop could conceivably allow that loop to be re-written so that the RC could be ignored, but doing so sounds like more work than should go in this PR.

I think you should ignore this for now. The simplest way forward is to accept that invalid controllers will not be handled—you just don't want them to impact availability for the correct controllers.  I think you can get that easily by returning a closed `updated` and `errors`.